src/Conversion/ONNXToTOSA/Tensor/Split.cpp

```diff
@@ -0,0 +1,82 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===------------- Split.cpp - Split Op---------===//
+//
+// Copyright (c) 2023 Advanced Micro Devices, Inc.
+//
+// =============================================================================
+//
+// This file lowers ONNX SplitOp operator to TOSA dialect.
+//
+//===----------------------------------------------------------------------===//
+
+#include "src/Conversion/ONNXToTOSA/DialectBuilder.hpp"
+#include "src/Conversion/ONNXToTOSA/ONNXToTOSACommon.hpp"
+#include "src/Conversion/ONNXToTOSA/ONNXToTOSALegalizeUtils.hpp"
+#include "src/Dialect/ONNX/ONNXOps.hpp"
+
+using namespace mlir;
+namespace onnx_mlir {
+namespace {
+class ONNXSplitOpLoweringToTOSA : public OpConversionPattern<ONNXSplitOp> {
+public:
+  using OpConversionPattern::OpConversionPattern;
+  using OpAdaptor = typename ONNXSplitOp::Adaptor;
+  LogicalResult matchAndRewrite(ONNXSplitOp op, OpAdaptor adaptor,
+      ConversionPatternRewriter &rewriter) const override {
+    Value input = adaptor.getInput();
+    if (!onnx_mlir::hasShapeAndRank(input)) {
```

Comment:
I think you can simplify this by just checking `hasStaticShape`.  And remove the code of the staticshape below.
```suggestion
    if (!hasStaticShape(inputType)) {
      return rewriter.notifyMatchFailure(op, "only static shapes are supported");
    }
```

Reply:
`hasStaticShape` does essentially the same thing as `hasShapeAndRank` and also checks that it is static.